File: Sum/RunningSum/Java/Solution2.java
package Sum.RunningSum.Java;

public class Solution2 {

    // O(n) time and space complexity
    public int[] runningSum(int[] nums) {
        int[] sums = new int[nums.length];
        sums[0] = nums[0];     
        
        for (int i = 1; i < nums.length; i++) {
            sums[i] = sums[i-1] + nums[i];
        }
        
        return sums;
    }
}

File: Sum/RunningSum/Kotlin/Solution2.kt
class Solution2 {
    // O(n) time complexity
    // O(1) space complexity, no new array is created
    fun runningSum(nums: IntArray): IntArray {

        for (i in 1 until nums.size) {
            nums[i] += nums[i - 1] 
        }
        
        return nums
    }
}